sdk/typescript/src/common/graphql/client.ts

```diff
@@ -22,7 +22,18 @@ const createFetchWithTimeout =
     }, timeout)
 
     try {
-      return (await fetch(input as NodeFetchRequestInfo, {
+      // Edge case for Deno that doesn't work as expected with node-fetch and would
+      // rather rely on its native fetch implementation
+      // See: https://github.com/dagger/dagger/issues/10546
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if (typeof (globalThis as any).Deno !== "undefined") {
```

Comment:
Maybe split this condition out to something like `isDeno` or something? Just for readability?